docs/api.md

```diff
@@ -594,7 +594,7 @@ Navigate to the next page in history.
 #### page.goto(url, options)
 - `url` <[string]> URL to navigate page to. The url should include scheme, e.g. `https://`.
 - `options` <[Object]> Navigation parameters which might have the following properties:
-  - `timeout` <[number]> Maximum navigation time in milliseconds, defaults to 30 seconds.
+  - `timeout` <[number]> Maximum navigation time in milliseconds, defaults to 30 seconds, pass `0` to disable timeout.
```

Comment:
There's a bunch of other methods that rely on NavigationWatcher, so you should add this clarification there as well:

```js
page.goForward()
page.boBack();
page.reload();
page.waitForNavigation()
```